### ISSUE TITLE
Not sure if frameLag calc is ideal

### ISSUE THREAD

**Issue by isaacl** [lichess-org/lila-ws#395]
frameLag is currently a [weighted running avg](https://github.com/lichess-org/lila-ws/blob/0655581989f88ef427de59d611dd1e1792423519/src/main/scala/Lag.scala#L27-L35) from trusted pongs.

This is very reasonable but what happens if a user hits a massive spike?  The weighted avg skyrockets and won't return for a while.  This translates into the old quota gain behavior for multiple moves in a row.

Consider:
- capping millis so it can't skyrocket the frameLag weighted avg
- changing the formula to drop faster if recording values that are lower than the current weighted avg.

For example:
```scala

private val trustedSpikeRefreshFactor = 0.05f
private val trustedNormalRefreshFactor = 0.1f
private val maxMillis = 5000
...

.fold(millis) { prev => {
    val weight = if (millis < prev) trustedNormalRefreshFactor else trustedSpikeRefreshFactor
    val cappedMillis = millis atMost maxMillis
    (prev * (1 - weight) + cappedMillis * weight).toInt
}
}
```

@ornicar @niklasf

**Comment by isaacl** [comment]
I'll do some digging and follow up w a PR

**Comment by tdball8444** [comment]
https://github.com/lichess-org/lila-ws/issues/395

**Comment by tdball8444** [comment]
private val trustedSpikeRefreshFactor = 0.05f
private val trustedNormalRefreshFactor = 0.1f
private val maxMillis = 5000
...

.fold(millis) { prev => {
    val weight = if (millis < prev) trustedNormalRefreshFactor else trustedSpikeRefreshFactor
    val cappedMillis = millis atMost maxMillis
    (prev * (1 - weight) + cappedMillis * weight).toInt
}
}

**Comment by isaacl** [comment]
whoops -- this issue isn't fully solved by #917 anymore, reopening

**Comment by philyawj** [comment]
> whoops -- this issue isn't fully solved by [#917](https://github.com/lichess-org/lila-ws/pull/917) anymore, reopening

Good catch. In the PR we ended up going for a simplified approach: https://github.com/lichess-org/lila-ws/pull/917#issuecomment-5591650179
